src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -74,6 +79,63 @@ impl Coordinator {
         .await;
     }
 
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub(crate) async fn sequence_peek_copy_to(
+        &mut self,
+        ctx: ExecuteContext,
+        plan::CopyToPlan {
+            select_plan,
+            desc,
+            to,
+            connection,
+            format_params,
+        }: plan::CopyToPlan,
+        target_cluster: TargetCluster,
+    ) {
+        let eval_uri = |to: HirScalarExpr| -> Result<Uri, AdapterError> {
```

Comment:
I think it's probably better to do that in the `plan_copy_to` stage and produce a `CopyToPlan` that has `uri: http::Uri`.

Reply:
> We had chatted about doing the resolution while sequencing

Ah, OK - @mjibson is the authority for this piece of the code, so if he recommended this go with that.

Reply:
I'm fine with this being wherever makes the most sense. One glance a while ago seemed like it was a bad fit inside the optimizer directory (that was many revisions ago). Things have probably changed since then and with optimizer folks now present, I don't care at all where this goes. It certainly doesn't need to be in the sequencer.

Reply:
In even an earlier (🙈) version (one of the commits in https://github.com/MaterializeInc/materialize/pull/24355) it was in the `plan_copy_to` in `dml.rs`. Should it be moved back there or okay to have this in the sequencer?

Reply:
I don't have an opinion on that so leave it where it is. If we find motivation to put it in a specific place we can move it later.